### ISSUE TITLE
[Bug]:@use-funnel/next generates an error when the browser is refreshed at a location other than initialStep.

### ISSUE THREAD

**Issue by XionWCFM** [toss/use-funnel#63]
### Package Scope

@use-funnel/next

### Bug description

I believe this behavior is a bug.

This is because this phenomenon does not occur in other adapter libraries. (like @use-funnel/browser ... etc...)

Is this perhaps intended behavior?

```tsx
TypeError: Cannot read properties of undefined (reading 'step')

webpack-internal:/node_modules/.pnpm/@use-funnel+core@0.0.5_react@18.3.1/node_modules/@use-funnel/core/dist/index.js (302:26)
```

If you check the call stack, the problem occurs at 302:26 in index.js.

```tsx
    const step = useMemo2(() => {
      const validContext = parseStepContext(currentState.step, currentState.context);
      return {
        ...validContext == null ? optionsRef.current.initial : {
          step: currentState.step,
          context: validContext
        },
        history,
        index: router.currentIndex,
        historySteps: router.history
      };
    }, [currentState, history, router.history, router.currentIndex, parseStepContext]);
```

### Expected behavior

We want to access steps without errors even when refreshing.

### To Reproduce

Environment 

- next 14.2.13
- @use-funnel/next 0.0.5

To reproduce this phenomenon, you can write the following simple code.

```tsx
import { useFunnel } from "@use-funnel/next";

export default function Home() {
	const funnel = useFunnel<{
		start: Record<string, unknown>;
		end: Record<string, unknown>;
	}>({
		id: "hello",
		initial: {
			step: "start",
			context: {},
		},
	});

	return (
		<div className="">
			<funnel.Render
				start={({ history }) => (
					<div>
						start step
						<button type="button" onClick={() => history.push("end")}>
							next
						</button>
					</div>
				)}
				end={() => <div>end step</div>}
			/>
		</div>
	);
}

```

This will be reproduced if you click the next button and try refreshing the browser when you reach the end step.

### Possible Solution

This is where I have doubts.

```tsx
export const useFunnel = createUseFunnel<NextPageRouteOption>(({ id, initialState }) => {
  const router = useRouter();
```

next/router accesses searchParams through isReady as if it were processed through useEffect.

In other words, when isReady is false, assets such as searchParams cannot be accessed.

Try running the following code

```tsx
import { useRouter } from "next/router";

export default function Home() {
	const router = useRouter();
	console.log(`router ready ${router.isReady} ${router.query.hello}`);
	return <div className="">hello</div>;
}

```

Then turn on the development server and access the path.

```
http://localhost:3000/?hello=hel
```

If you run this code, console.log runs a total of three times in the development environment.

1. Once on serverside

router erady false undefined

2. Once on clientside

router erady false undefined

3. Once on clientside

router ready true hel

I think the error occurs at this point because the query is undefined on the server side and client side.

I haven't read the entire code base, so I'm not sure if this is exactly the cause of the error.

If this is the cause, you may want to consider these solutions:

I think [toss already created a useRouter that triggers suspense to solve this problem](https://github.com/toss/slash/blob/main/packages/react/use-query-param/src/hooks/useNextRouter.ts)

### etc.

🔥 thank you for your hard work

**Comment by minuukang** [comment]
Thanks for find issue and solution! I'll apply it quickly.

**Comment by minuukang** [comment]
I find a bug when refresh a page in `@use-funnel/next`. it should not keep the history state 😢.

So we've made an emergency patch to get the `initial` state when this happens (#65), and we'll work with contributors to figure out how to maintain it.

**Comment by minuukang** [comment]
This issue is solved from @use-funnel/next@0.0.7! Thanks for your issue ❤️

**Comment by minuukang** [comment]
0.0.7 version contained a part that used an incorrect value, which was promptly fixed in "0.0.8" 🙇🏽
